scrapers/ma/bills.py

```diff
@@ -68,91 +70,55 @@ def scrape(
+            yield from self.scrape_chamber("upper", session)
         else:
-            yield from self.scrape_chamber(chamber, session, sort, page_limit)
+            yield from self.scrape_chamber(chamber, session)
 
-    def scrape_chamber(self, chamber, session, sort=None, page_limit=None):
-        # for the chamber of the action
-
-        # Pull the search page to get the filters
-        search_url = "https://malegislature.gov/Bills/Search?SearchTerms=&Page=1"
-        page = lxml.html.fromstring(self.get(search_url, verify=False).text)
-        self.session_filters = self.get_refiners(page, "lawsgeneralcourt")
-        self.chamber_filters = self.get_refiners(page, "lawsbranchname")
+    def scrape_bill_list(self, session):
+        session_numeric = re.sub(r"[^0-9]", "", session)
+        # note -- this returns XML to a browser, but json to curl/python
+        api_url = (
+            f"https://malegislature.gov/api/GeneralCourts/{session_numeric}/Documents"
+        )
 
-        if page_limit:
-            last_page = page_limit
-        else:
-            last_page = self.get_max_pages(session, chamber)
-
-        for pageNumber in range(1, last_page + 1):
-            bills = self.list_bills(session, chamber, pageNumber, sort)
-            for bill in bills:
-                bill = self.format_bill_number(bill).replace(" ", "")
-                yield from self.scrape_bill(session, bill, chamber)
-
-    def list_bills(self, session, chamber, pageNumber, sort=None):
-        session_filter = self.session_filters[session]
-        chamber_filter = self.chamber_filters[self.chamber_map[chamber]]
-
-        if sort is None:
-            search_url = (
-                "https://malegislature.gov/Bills/Search?"
-                "SearchTerms="
-                "&Page={}"
-                "&SortManagedProperty=lawsbillnumber"
-                "&Direction=asc"
-                "&Refinements%5Blawsgeneralcourt%5D={}"
-                "&Refinements%5Blawsbranchname%5D={}".format(
-                    pageNumber, session_filter, chamber_filter
-                )
-            )
-        elif sort == "latest":
-            search_url = (
-                "https://malegislature.gov/Bills/Search?"
-                "SearchTerms=&Page={}&Refinements%5Blawsgeneralcourt%5D={}"
-                "&Refinements%5Blawsbranchname%5D={}".format(
-                    pageNumber, session_filter, chamber_filter
+        list_data = self.get(api_url, verify=False).content
+        for row in json.loads(list_data):
+            if "H" in str(row["BillNumber"]) or "H" in str(row["DocketNumber"]):
+                chamber = "lower"
+            elif "S" in str(row["BillNumber"]) or "S" in str(row["DocketNumber"]):
+                chamber = "upper"
+            else:
+                self.error(
+                    f"Unknown bill type - bill {row['BillNumber']} - docket {row['DocketNumber']}"
                 )
+
+            self.bill_list[chamber].append(
```

Comment:
This object feels a little weird. Should we just have a dict instead of a list of dicts?

Reply:
I think since I was adapting the old code (which is chamber based) I had it in my head that there was some reason to split it, but there's really not; good idea i'll refactor that out.